egs2/TEMPLATE/vc1/vc.sh

```diff
@@ -0,0 +1,830 @@
+    log "Error: only supported: --feats_type raw"
+    exit 2
+fi
+
+# Set tag for naming of model directory
+if [ -z "${tag}" ]; then
+    if [ -n "${train_config}" ]; then
+        tag="$(basename "${train_config}" .yaml)_${feats_type}"
+    else
+        tag="train_${feats_type}"
+    fi
+    # Add overwritten arg's info
+    if [ -n "${train_args}" ]; then
+        tag+="$(echo "${train_args}" | sed -e "s/--/\_/g" -e "s/[ |=]//g")"
+    fi
+fi
+if [ -z "${inference_tag}" ]; then
+    if [ -n "${inference_config}" ]; then
+        inference_tag="$(basename "${inference_config}" .yaml)"
+    else
+        inference_tag=inference
+    fi
+    # Add overwritten arg's info
+    if [ -n "${inference_args}" ]; then
+        inference_tag+="$(echo "${inference_args}" | sed -e "s/--/\_/g" -e "s/[ |=]//g")"
+    fi
+    inference_tag+="_$(echo "${inference_model}" | sed -e "s/\//_/g" -e "s/\.[^.]*$//g")"
+fi
+
+# The directory used for collect-stats mode
+if [ -z "${vc_stats_dir}" ]; then
+    vc_stats_dir="${expdir}/vc_stats_${feats_type}"
+    if [ "${feats_extract}" != fbank ]; then
+        vc_stats_dir+="_${feats_extract}"
+    fi
+fi
+# The directory used for training commands
+if [ -z "${vc_exp}" ]; then
+    vc_exp="${expdir}/vc_${tag}"
+fi
+
+
+# ========================== Main stages start from here. ==========================
+
+if ! "${skip_data_prep}"; then
+    if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+        log "Stage 1: Data preparation for data/*"
+        # [Task dependent] Need to create data.sh for new corpus
+        local/data.sh ${local_data_opts}
+    fi
+
+
+    if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+        log "Stage 2: Format wav.scp: data/ -> ${data_feats}/"
+        for dset in "${src_train_set}" "${src_valid_set}" ${src_test_sets} "${trg_train_set}" "${trg_valid_set}" ${trg_test_sets}; do
+            if [ "${dset}" = "${src_train_set}" ] || [ "${dset}" = "${src_valid_set}" ] || [ "${dset}" = "${trg_train_set}" ] || [ "${dset}" = "${trg_valid_set}" ]; then
+                _suf="/org"
+            else
+                _suf=""
+            fi
+            utils/copy_data_dir.sh data/"${dset}" "${data_feats}${_suf}/${dset}"
+            rm -f ${data_feats}${_suf}/${dset}/{segments,wav.scp,reco2file_and_channel}
+            _opts=
+            if [ -e data/"${dset}"/segments ]; then
+                _opts+="--segments data/${dset}/segments "
+            fi
+            # shellcheck disable=SC2086
+            scripts/audio/format_wav_scp.sh --nj "${nj}" --cmd "${train_cmd}" \
+                --audio-format "${audio_format}" --fs "${fs}" ${_opts} \
+                "data/${dset}/wav.scp" "${data_feats}${_suf}/${dset}"
+            echo "${feats_type}" > "${data_feats}${_suf}/${dset}/feats_type"
+        done
+
+        # NOTE(peter): skipped Extract X-vector
+
+        # NOTE(peter): skipped Prepare spk id input
+
+        # NOTE(peter): skipped Prepare lang id input
```

Comment:
It is better to show messages.
```suggestion
        # NOTE(peter): skipped Extract X-vector
        if "${use_xvector}"; then
            log "[ERROR] X-vector is not supported yet."
            exit 1
        fi
        # NOTE(peter): skipped Prepare spk id input
        if "${use_sid}"; then
            log "[ERROR] Speaker ID is not supported yet."
            exit 1
        fi
        # NOTE(peter): skipped Prepare lang id input
        if "${use_lid}"; then
            log "[ERROR] Language ID is not supported yet."
            exit 1
        if
```